public/app/features/alerting/unified/components/rules/RulesTable.tsx

```diff
@@ -109,18 +109,25 @@ function useColumns(showSummaryColumn: boolean, showGroupColumn: boolean, showNe
   const { hasRuler, rulerRulesLoaded } = useHasRuler();
 
   return useMemo((): RuleTableColumnProps[] => {
+    const ruleIsDeleting = (rule: CombinedRule) => {
+      const { namespace, promRule, rulerRule } = rule;
+      const { rulesSource } = namespace;
+      return Boolean(hasRuler(rulesSource) && rulerRulesLoaded(rulesSource) && promRule && !rulerRule);
```

Comment:
is not the opposite? if rule is being deleted, we still have the ruler rule, but not the promRule?

Reply:
I was initially confused too but the logic is correct here.

A rule is being "deleted" when the prometheus endpoint still tracks state for it but it no longer exists in the ruler API (the definition has been removed).

Reply:
you're right.... as the ruler is faster than prometheus. But this text was confusing me on the description of the PR:

> In Cloud instances, there can be a lag of ~1minute on the ruler API.